### PR TITLE
Add deprecation warnings.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: sdk/nodejs/runtime/closure/codePaths.ts [hunk 1]
```diff
@@ -87,6 +89,7 @@ export async function computeCodePaths(
 
     let options: CodePathOptions;
     if (Array.isArray(optionsOrExtraIncludePaths)) {
+        log.warn("'function computeCodePaths(string[])' is deprecated. Use the [computeCodePaths] overload that takes a [CodePathOptions] instead.");
```
Comment (CyrusNajmabadi) [inline sdk/nodejs/runtime/closure/codePaths.ts]: general idea.  Name the symbol and say it's deprecated (in this case, i need to include the `string[]` portion as this is an overload).  Then have a message saying what to do.  The message here matches the messae in the `@deprecated` message above.

File: sdk/nodejs/runtime/closure/serializeClosure.ts [hunk 2]
```diff
@@ -94,12 +95,13 @@ export async function serializeFunction(
...
 
 /**
- * @deprecated This function has been replaced by `serializeFunction`, which accepts additional parameters and returns
- * more details about the serialized function.  This form will be removed in a future release of this package.
+ * @deprecated Please use 'serializeFunction' instead."
  */
 export async function serializeFunctionAsync(
         func: Function,
         serialize?: (o: any) => boolean): Promise<string> {
+    log.warn("'function serializeFunctionAsync' is deprecated.  Please use 'serializeFunction' instead.");
```
Comment (CyrusNajmabadi) [inline sdk/nodejs/runtime/closure/serializeClosure.ts]: same pattern.  in teh log message, saying the symbol that is deprecated.  then say what to do.  The latter sentence should match what is in `@deprecated`.   We can keep things simple.

File: sdk/nodejs/runtime/closure/serializeClosure.ts [hunk 3]
```diff
@@ -94,12 +95,13 @@ export async function serializeFunction(
 }
 
 /**
- * @deprecated This function has been replaced by `serializeFunction`, which accepts additional parameters and returns
- * more details about the serialized function.  This form will be removed in a future release of this package.
+ * @deprecated Please use 'serializeFunction' instead."
```
Comment (lukehoban) [inline sdk/nodejs/runtime/closure/serializeClosure.ts]: Trailing `"`?

**Comment by lukehoban** [comment]
Let's also proactively go clean up the existing users of these (at least within the `pulumi` org):

https://github.com/search?q=org%3Apulumi+serializeFunction&type=Code
https://github.com/search?q=org%3Apulumi+computeCodePaths&type=Code

File: sdk/nodejs/runtime/closure/serializeClosure.ts (hunk 3 above)
Comment (CyrusNajmabadi) [inline sdk/nodejs/runtime/closure/serializeClosure.ts]:
```suggestion
 * @deprecated Please use 'serializeFunction' instead.
```

File: sdk/nodejs/runtime/closure/codePaths.ts [hunk 4]
```diff
@@ -54,25 +55,26 @@ export interface CodePathOptions {
...
-// See [CodePathOptions] for information on ways to control and configure the final set of paths
-// included in the resultant asset/archive map.
-//
-// Note: this functionality is specifically intended for use by downstream library code that is
-// determining what is needed for a cloud-lambda.  i.e. the aws.serverless.Function or
-// azure.serverless.FunctionApp libraries.  In general, other clients should not need to use this
-// helper.
-
+/**
+ * computeCodePaths computes the local node_module paths to include in an uploaded cloud 'Lambda'.
```
Comment (metral) [inline sdk/nodejs/runtime/closure/codePaths.ts]: This is not marked with `@deprecated`, yet we warn in the log below. Is this missing?

**Comment by CyrusNajmabadi** [comment]
> Let's also proactively go clean up the existing users of these (at least within the pulumi org):

Will do.

File: sdk/nodejs/runtime/closure/codePaths.ts (hunk 4 above)
Comment (CyrusNajmabadi) [reply sdk/nodejs/runtime/closure/codePaths.ts]: no.  This is an overload.  You won't get the warning unless you call through the *array* argument codepath.

**Comment by CyrusNajmabadi** [comment]
> Let's also proactively go clean up the existing users of these (at least within the pulumi org):

Have audited and fixed up the ones taht need fixing.

**Pull request merged**